src/Agent.Service/Linux/AgentService.ts

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env node
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/// <reference path="./definitions/node.d.ts"/>
+
+import childProcess = require("child_process");
+import os = require("os")
+import fs = require("fs")
+import path = require("path")
+import traceModule = require("./trace")
+
+var supported = ['linux']
+
+if (supported.indexOf(process.platform) == -1) {
+    console.log('Unsupported platform: ' + process.platform);
+    console.log('Supported platforms are: ' + supported.toString());
+    process.exit(1);
+}
+
+var stopping:boolean = false;
+var listener:childProcess.ChildProcess = null;
+var trace:traceModule.ITraceWriter = traceModule.getAgentServiceDiagnosticWriter();
+
+var runService = function() {
+    var listenerExePath = path.join(__dirname, 'Agent.Listener');
+    trace.info('Starting Agent listener');
+
+    if(!stopping) {
+	try {
+	    listener = childProcess.spawn(listenerExePath, ['run'], { env: process.env });
+	    trace.info('started listener process');
+		
+	    listener.stdout.on('data', (data) => {
+		trace.info(data);
+	    });
+
+	    listener.stderr.on('data', (data) => {
+		trace.error(data);
+	    });
+
+	    listener.on('close', (code, signal) => {
+		trace.info(`Agent listenere exited with error code ${code}`);
+
+		if (code == 1)
+		{
+		    trace.error('Can not start agent listener process, check logs for error detail');
```

Comment:
We may want to console.log here as well.  I know that at least on OSX all stdout and stderr out goes to the service (console) log.